src/persist-client/src/impl/state.rs

```diff
@@ -150,18 +150,18 @@ where
     }
 
     fn update_since(&mut self) {
-        let mut readers = self.readers.values();
-        let mut since = match readers.next() {
-            Some(reader) => reader.since.clone(),
-            None => {
-                // If there are no current readers, leave `since` unchanged so
-                // it doesn't regress.
-                return;
-            }
-        };
-        while let Some(reader) = readers.next() {
+        if self.readers.is_empty() {
```

Comment:
I find the new impl *less* simple :). it requires reasoning about what the "identity input" of meet is, whereas the old impl didn't

Reply:
removed!

Reply:
removed!